src/python/pants/base/exception_sink.py

```diff
@@ -154,22 +148,23 @@ def reset_interactive_output_stream(cls, interactive_output_stream):
     OS state:
     - Overwrites the SIGUSR2 handler.
 
-    This is where the the error message on exit will be printed to as well.
+    This method registers a SIGUSR2 handler, which permits a non-fatal `kill -31 <pants pid>` for
+    stacktrace retrieval. This is also where the the error message on fatal exit will be printed to.
     """
+    logger.debug('registering a SIGUSR2 handler')
+
     try:
       # NB: mutate process-global state!
-      if faulthandler.unregister(signal.SIGUSR2):
-        logger.debug('re-registering a SIGUSR2 handler')
       # This permits a non-fatal `kill -31 <pants pid>` for stacktrace retrieval.
       faulthandler.register(signal.SIGUSR2, interactive_output_stream,
                             all_threads=True, chain=False)
-
-      # NB: mutate the class variables!
-      # We don't *necessarily* need to keep a reference to this, but we do here for clarity.
-      cls._interactive_output_stream = interactive_output_stream
     except ValueError:
-      # Warn about "ValueError: IO on closed file" when stderr is closed.
-      ExceptionSink.log_exception("Cannot reset output stream - sys.stderr is closed")
+      # Warn about "ValueError: IO on closed file" when the stream is closed.
+      cls.log_exception(
+        "Cannot reset interactive_output_stream -- stream (probably stderr) is closed")
+
+    # NB: mutate the class variables!
```

Comment:
(1) The whole original message is now incorrect, as we now use that variable elsewhere in `.exit_with_failure()`, instead of just to handle `SIGUSR2`.
(2) I was actually intending to avoid updating the output stream class variable if an exception was raised, so I have moved it back into the `try` block.